lib/internal/util/parse_args/parse_args.js

```diff
@@ -0,0 +1,315 @@
+'use strict';
+
+const {
+  ArrayPrototypeForEach,
+  ArrayPrototypePushApply,
+  ArrayPrototypeShift,
+  ArrayPrototypeSlice,
+  ArrayPrototypePush,
+  ArrayPrototypeUnshiftApply,
+  ObjectPrototypeHasOwnProperty: ObjectHasOwn,
+  ObjectEntries,
+  StringPrototypeCharAt,
+  StringPrototypeIncludes,
+  StringPrototypeIndexOf,
+  StringPrototypeSlice,
+} = primordials;
+
+const {
+  validateArray,
+  validateObject,
+  validateString,
+  validateUnion,
+  validateBoolean,
+} = require('internal/validators');
+
+const {
+  findLongOptionForShort,
+  isLoneLongOption,
+  isLoneShortOption,
+  isLongOptionAndValue,
+  isOptionValue,
+  isOptionLikeValue,
+  isShortOptionAndValue,
+  isShortOptionGroup,
+  objectGetOwn,
+  optionsGetOwn,
+} = require('internal/util/parse_args/utils');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_VALUE,
+    ERR_PARSE_ARGS_INVALID_OPTION_VALUE,
+    ERR_PARSE_ARGS_UNKNOWN_OPTION,
+    ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL,
+  },
+} = require('internal/errors');
+
+function getMainArgs() {
+  // This function is a placeholder for proposed process.mainArgs.
+  // Work out where to slice process.argv for user supplied arguments.
+
+  // Electron is an interesting example, with work-arounds implemented in
+  // Commander and Yargs. Hopefully Electron would support process.mainArgs
+  // itself and render this work-around moot.
+  //
+  // In a bundled Electron app, the user CLI args directly
+  // follow executable. (No special processing required for unbundled.)
+  // 1) process.versions.electron is either set by electron, or undefined
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
+  // 2) process.defaultApp is undefined in a bundled Electron app, and set
+  //    in an unbundled Electron app
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
```

Comment:
I think we use `HEAD` links only for repos that have `master` as the default branch name but since electron has already migrated to using `main`, I think it's fine if we use the link with the branch name. A tiny advantage of using links with branch names is that it allows you to edit the linked file (`HEAD` links don't allow you to edit the files).

Reply:
Using HEAD is less hassle if electron ever change the name of their branch again.

Reply:
only if the old branch name is deleted, and nobody's created a branch name that breaks the chain in the meantime.

In other words, the only reliable thing that survives renaming is to put "HEAD"; the redirects are just a best-effort convenience.

Reply:
(Arg. Casual copy of an URL in what might have been my first commit... 😆 )

Reply:
Actually, I think it would be better if we use the official doc link on electron's website
```suggestion
  //    see https://www.electronjs.org/docs/latest/api/process#processdefaultapp-readonly
```